### DISCUSSION TITLE
properties are showing up as optional, not required

### DISCUSSION THREAD

**Discussion by mlilback** [colinhacks/zod#1093] [General]
I'm zod a spin on a Typescript project in VS Code. I'm defining my type as
```
export const promoSchema = z.object({
  promotionId: z.number().nonnegative(),
  code: z.string().min(3),
  amount: z.preprocess((arg: string) => new Currency(arg), z.instanceof(Currency)),
  validFrom: z.preprocess((arg) => typeof arg == "string" ? new Date(arg) : undefined, z.date()),
  validUntil: z.preprocess((arg) => typeof arg == "string" ? new Date(arg) : undefined, z.date()),
})
export type PromoI = z.infer<typeof promoSchema>
```

When I hover over PromoI it shows the type as:
```
type PromoI = {
    promotionId?: number;
    code?: string;
    amount?: Currency;
    validFrom?: Date;
    validUntil?: Date;
}
```

Why are the properties all optional? I thought they are required by default?

**Comment by andrhamm** [comment]
Zod requires that you use TypeScript in strict mode in your tsconfig

https://github.com/colinhacks/zod#requirements

**Reply by mlilback** [reply]
I enabled strict mode and still have the same problem.

**Reply by nicolasembleton** [reply]
That solved it indeed... Opened up a *lot* of other issues however. 😅

**Comment by nicolasembleton** [comment]
@mlilback you're probably way past that, but wondering if you meant to express that these fields were required for the input (parse) such as `z.input<typeof promoSchema>` ? I'm asking because I know I did and this could help others that read this thread.
